cli_tools/gce_ovf_export/exporter/exporter.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging/service"
+	storageutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/storage"
+	ovfexportdomain "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_export/domain"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	daisycompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"github.com/GoogleCloudPlatform/compute-image-tools/proto/go/pb"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+)
+
+const (
+	logPrefix  = "[ovf-export]"
+	bytesPerGB = int64(1024 * 1024 * 1024)
+)
+
+const (
+	//Alpha represents alpha release track
+	Alpha = "alpha"
+
+	//Beta represents beta release track
+	Beta = "beta"
+
+	//GA represents GA release track
+	GA = "ga"
+)
+
+// OVFExporter is responsible for exporting GCE VMs/GMIs to OVF/OVA
+type OVFExporter struct {
+	storageClient          domain.StorageClientInterface
+	computeClient          daisycompute.Client
+	mgce                   domain.MetadataGCEInterface
+	bucketIteratorCreator  domain.BucketIteratorCreatorInterface
+	Logger                 logging.LoggerInterface
+	instanceDisksExporter  ovfexportdomain.InstanceDisksExporter
+	instanceExportPreparer ovfexportdomain.InstanceExportPreparer
+	instanceExportCleaner  ovfexportdomain.InstanceExportCleaner
+	params                 *ovfexportdomain.OVFExportParams
+
+	// BuildID is ID of Cloud Build in which this OVF export runs in
+	BuildID string
+
+	inspector                 commondisk.Inspector
+	ovfDescriptorGenerator    ovfexportdomain.OvfDescriptorGenerator
+	manifestFileGenerator     ovfexportdomain.OvfManifestGenerator
+	exportedDisks             []*ovfexportdomain.ExportedDisk
+	bootDiskInspectionResults *pb.InspectionResults
+	traceLogs                 []string
+	loggableBuilder           *service.OvfExportLoggableBuilder
+}
+
+// NewOVFExporter creates an OVF exporter, including automatically populating dependencies,
+// such as compute/storage clients.
+func NewOVFExporter(params *ovfexportdomain.OVFExportParams) (*OVFExporter, error) {
+	ctx := context.Background()
+	log.SetPrefix(logPrefix + " ")
+	logger := logging.NewStdoutLogger(logPrefix)
+	storageClient, err := storageutils.NewStorageClient(ctx, logger)
+	if err != nil {
+		return nil, err
+	}
+	computeClient, err := createComputeClient(&ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	metadataGCE := &computeutils.MetadataGCE{}
+
+	paramValidator := NewOvfExportParamValidator(computeClient)
+	paramPopulator := NewPopulator(metadataGCE, storageClient,
+		storageutils.NewResourceLocationRetriever(metadataGCE, computeClient),
+		storageutils.NewScratchBucketCreator(ctx, storageClient),
+	)
+	if err := ValidateAndPopulateParams(params, paramValidator, paramPopulator); err != nil {
+		return nil, err
+	}
```

Comment:
I've refactored flags into args to support trimming. I'll leave validation/population consolidation for later time as it would be a bigger effort right now.